MovieLibrary/src/com/company/service/UserService.java

```diff
@@ -0,0 +1,154 @@
+package com.company.service;
+
+
+import com.company.User;
+import com.company.UserRole;
+
+import java.io.FileNotFoundException;
+import java.util.List;
+
+
+import static com.company.FileUtils.saveProgress;
+import static com.company.IOUtils.readIntFromUser;
+import static com.company.IOUtils.readStringFromUser;
+import static com.company.Utils.*;
+
+public class UserService {
+
+    private static final String ADMIN_USERNAME = "admin";
+    private static final String GUEST_USERNAME = "guest";
+    public static UserRole getRole(User user) {
+
+        return user.getLogin().equals(ADMIN_USERNAME) ? UserRole.ADMIN : (user.getLogin().equals(GUEST_USERNAME) ? UserRole.GUEST : UserRole.USER);
+    }
+
+    public static boolean  isAdmin(User user) {
+
+        return getRole(user).equals(UserRole.ADMIN);
+    }
+    public static boolean  isGuest(User user) {
+
+        return getRole(user).equals(UserRole.GUEST);
+    }
+
+    public static User loginOrRegisterSwitch(List<User> users) {
+            User user;
+            int x;
+            do {
+                showWelcomeMenu();
+                x = readIntFromUser("option");
+                switch (x) {
+                    case 1:
+                        user = login(users);
+                        return user;
+                    case 2:
+                        try {
+                            user = addUser(users);
+                            try{
+                                saveProgress(users,"users.dat");
+                            } catch (FileNotFoundException e) {
+                                System.out.println(e.getMessage());
+                            }
+                            return user;
+                        } catch (Exception e) {
+                            System.out.println(e.getMessage());
+                        }
+                        break;
+                    case 3:
+                        user = new User(0,"guest","guest");
+                        return user;
+                    case 9:
+                        break;
+                    default:
+                        System.out.println("Enter a number corresponding to selected action");
+                }
+            } while (x!=1 && x!=2 && x!=9);
+            return new User(0,"guest","guest");
+            }
+
+    public static User login(List<User> users) {

```

Comment:
jezeli metoda ma więcej niż 2 wcięcia to znaczy, że zaczyna się "code-smell", pewnie warto rozważyć tu wydzielenie metod prywatnych, które go uproszczą i rozbicie tych ifów.